package.json

```diff
@@ -24,5 +24,9 @@
   "packageManager": "yarn@4.4.1",
   "devDependencies": {
     "syncpack": "^12.4.0"
+  },
+  "resolutions": {
+    "@types/react": "^18",
+    "@types/react-dom": "^18"
```

Comment:
https://yarnpkg.com/configuration/manifest#resolutions

We were getting multiple versions of `@types/react` resolved by yarn. This can cause typescript problems. In particular, it was causing the error 
```
[ol-components]: Process started
[ol-components]: src/components/Carousel/Carousel.tsx(34,28): error TS2769: No overload matches this call.
[ol-components]:   Overload 1 of 6, '(component: ComponentClass<Settings, any>, options?: StyledOptions<Settings> | undefined): CreateStyledComponent<Settings & { ...; }, {}, { ...; }>', gave the following error.
[ol-components]:     Argument of type 'typeof Slider' is not assignable to parameter of type 'ComponentClass<Settings, any>'.
[ol-components]:       Types of property 'contextType' are incompatible.
[ol-components]:         Type 'import("/Users/cchudzicki/dev/mit-open/node_modules/@types/react-transition-group/node_modules/@types/react/index").Context<any> | undefined' is not assignable to type 'React.Context<any> | undefined'.
[ol-components]:           Type 'import("/Users/cchudzicki/dev/mit-open/node_modules/@types/react-transition-group/node_modules/@types/react/index").Context<any>' is not assignable to type 'React.Context<any>'.
[ol-components]:             Types of property 'Provider' are incompatible.
[ol-components]:               Type 'import("/Users/cchudzicki/dev/mit-open/node_modules/@types/react-transition-group/node_modules/@types/react/index").Provider<any>' is not assignable to type 'React.Provider<any>'.
[ol-components]:                 Types of parameters 'props' and 'props' are incompatible.
[ol-components]:                   Type 'React.ProviderProps<any>' is not assignable to type 'import("/Users/cchudzicki/dev/mit-open/node_modules/@types/react-transition-group/node_modules/@types/react/index").ProviderProps<any>'.
[ol-components]:                     Types of property 'children' are incompatible.
[ol-components]:                       Type 'React.ReactNode' is not assignable to type 'import("/Users/cchudzicki/dev/mit-open/node_modules/@types/react-transition-group/node_modules/@types/react/index").ReactNode'.
[ol-components]:                         Type 'bigint' is not assignable to type 'ReactNode'.
```

A similar issue cropped up in mitxonline a while ago. See
- https://github.com/mitodl/mitxonline/pull/2042#issuecomment-1932752978

for more details.

### Note:
You can fix this temporarily with `yarn dedupe @types/react`, but subsequent updates to `yarn.lock` might get you two different versions again.

This should fix it permanently (or at least until React 19 comes out, when we'll need to update the type verion).

I added a resolution for `@types/react-dom`, too, since the two go hand in hand.